### ISSUE TITLE
Can't load installed xontrib

### ISSUE THREAD

**Issue by seanfarley** [xonssh/xxh#34]
**Local OS (where xxh is installed):** macos
**Destination host OS:** ubuntu 18.04
**xxh version:**
**xxh-plugins installed:**

**Steps to Reproduce**
Whenever I try to run my own .xonshrc, and load some xontribs I always get an error that they're not installed:

```
$ xxh euclid
The following xontribs are enabled but not installed: 
   autojump
To install them run 
    xpip install xontrib-autojump

$ xpip install xontrib-autojump
$ logout
$ xxh euclid
$ The following xontribs are enabled but not installed: 
   autojump
To install them run 
    xpip install xontrib-autojump
```

It seems that wherever the packages are being installed to, it doesn't get source upon login.

**For community:**
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by anki-code** [comment]
Hey @sean!
Let's strart from simpler case (I will use xxh-dev environment):
```
root@start:/xxh-dev# xxh/xxh -i keys/id_rsa ubuntu_k
    ubuntu_k:/root/.xxh not found. Install xxh? [Y/n] 
    Install xxh to ubuntu_k:/root/.xxh
    ...
    First run xonsh on ubuntu_k
root@ubuntu_k ~ # 
                                                                                                                                                                          
root@ubuntu_k ~ # xpip install xontrib-autojump                                                                                                                                                         
Collecting xontrib-autojump
  Downloading xontrib-autojump-1.2.tar.gz (2.6 kB)
Installing collected packages: xontrib-autojump
    Running setup.py install for xontrib-autojump ... done
Successfully installed xontrib-autojump-1.2

root@ubuntu_k ~ # xontrib list autojump                                                                                                                                                                          
autojump            installed      not-loaded

root@ubuntu_k ~ # xontrib load autojump                                                                                                                                                                 
root@ubuntu_k ~ # xontrib list autojump                                                                                                                                                                 
autojump  installed      loaded

root@ubuntu_k ~ # exit

root@start:/xxh-dev# xxh/xxh -i keys/id_rsa ubuntu_k
root@ubuntu_k ~ # xontrib load autojump                                                                                                                                                                 
root@ubuntu_k ~ # xontrib list autojump                                                                                                                                                                           
autojump            installed      loaded
```
Install, load and load after logout is working.

Let's deep dive with your case. What are you try to do? Please describe step by step.

**Comment by anki-code** [comment]
Thank you for reporting! I can reproduce the problem with loading xontribs.
While I'm deep diving: as workaround you can try https://github.com/xxh/xxh-plugin-autojump
Also try new 0.3.1!

**Comment by anki-code** [comment]
When we try to load xontrib before installing then the xonsh "cached" the state and we get fail:
```
# xxh host
$ xontrib load autojump
autojump  not-installed  not-loaded
$ xpip install xontrib-autojump
$ xontrib load autojump
The following xontribs are enabled but not installed ...
$ xontrib list autojump
autojump  not-installed  not-loaded
```

But when we install xontrib before loading we get success:
```
# xxh host
$ xpip install xontrib-autojump
$ xontrib load autojump
$ xontrib list autojump
autojump  installed  loaded
```
I'll try to address it to xonsh team https://github.com/xonsh/xonsh/issues/3461.

**Comment by anki-code** [comment]
@seanfarley Meanwhile, I try again (xxh 0.3.1) and I can't reproduce your example exactly. 

If I install autojump via xpip and do relogin I can load autojump every time:
```
# xxh host
$ xpip install xontrib-autojump
$ exit
# xxh host
$ xontrib load autojump
$ # OK
```
Probably the key in what you are doing before xontrib loading.

Also as workaround I've found a some way using [xxh-plugins](https://github.com/xonssh/xxh-plugin-sample):
1. Create local `~/.xxh/plugnis/xxh-plugin-myjump`
2. Place file xonshrc.xsh
```
if not ($XXH_HOME / 'pip/xontrib/autojump.xsh').exists():
    xpip install xontrib-autojump
xontrib load autojump
```
3. Install to your host: `xxh host +if` (forced mode)
4. And voila!

**Comment by seanfarley** [comment]
Hmmm, ok, I'll read this more in-depth and try it out later today! Thanks for the quick followup :-)

**Comment by anki-code** [comment]
Update. I dive deeper and found the issue with xonsh xontribs and pip target - https://github.com/xonsh/xonsh/issues/3463

**Comment by anki-code** [comment]
@seanfarley 

~1. I've fix this by creating `xontrib-install` alias and release new xxh 0.3.4.~ (https://github.com/xxh/xxh/issues/34#issuecomment-593066069)

2. Also I've updated https://github.com/xxhxxh-plugin-autojump to use autojump xontrib more naturally.

**Comment by seanfarley** [comment]
Ah, cool! I'll try that out.

**Comment by anki-code** [comment]
@seanfarley sorry for fast speed ;) 

I've just released 0.3.5 with seamless way to avoid [xontrib issue](#issuecomment-592937419). 

Now you can use ordinary way to install xontribs using `xpip`. Just run:

```
python3 -m pip install --upgrade xxh-xxh
xxh host +if
xpip install xontrib-autojump
xontrib load autojump
```

Great weekend!

**Comment by seanfarley** [comment]
Yeah, looks like that works! Thanks!
